kv/union_store.go

```diff
@@ -106,10 +111,39 @@ func (us *UnionStore) Delete(k []byte) error {
 	return us.WBuffer.Set(k, nil)
 }
 
+// CheckExpect loads all expect values from store then checks if all values are matched.
+func (us *UnionStore) CheckExpect() error {
+	var keys []Key
+	for it := us.expect.NewIterator(nil); it.Valid(); it.Next() {
+		keys = append(keys, []byte(it.Key()))
+	}
+	if len(keys) == 0 {
+		return nil
+	}
+	values, err := us.Snapshot.BatchGet(keys)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for it := us.expect.NewIterator(nil); it.Valid(); it.Next() {
+		if len(it.Value()) == 0 {
```

Comment:
Not all the it.Value() in in expect is nil? 
For I see:
err = c.expect.Set(k, nil)